### PR TITLE
Parameterization of elu layer

### PR DESCRIPTION
**Pull request by srk97**
### Added the hyperparameter `alpha` to elu activation unit.

The modified forward propagation is:

elu

### CODE REVIEW

**Comment by srk97** [comment]
@edgarriba @Randl @karandesai-96 All checks have passed. Could you review it?

File: test/test.cpp [hunk 1]
```diff
@@ -41,7 +42,6 @@ using namespace tiny_dnn::activation;
 #include "test_slice_layer.h"
 #include "test_target_cost.h"
 #include "test_tensor.h"
```
Comment (beru) [inline test/test.cpp]: They are in alphabetical order now. No one knows what comes next.  :scream:

File: tiny_dnn/activations/elu_layer.h [hunk 2]
```diff
@@ -33,14 +50,15 @@ class elu_layer : public activation_layer {
                            const vec_t &dy) override {
     for (size_t j = 0; j < x.size(); j++) {
       // dx = dy * (gradient of elu)
-      dx[j] = dy[j] * (y[j] > float_t(0) ? float_t(1) : (float_t(1) + y[j]));
+      dx[j] = dy[j] * (y[j] > float_t(0) ? float_t(1) : (alpha_ + y[j]));
```
Comment (beru) [inline tiny_dnn/activations/elu_layer.h]: It seems correct according to below references.

https://arxiv.org/pdf/1511.07289.pdf
https://en.wikipedia.org/wiki/Activation_function#Comparison_of_activation_functions

Chainer's `ELU::backward_cpu` looks different though.
https://github.com/chainer/chainer/blob/master/chainer/functions/activation/elu.py
It multiplies with alpha. I don't know why...

Caffe's `ELULayer::Backward_cpu` uses addition, so it looks the same.
https://github.com/BVLC/caffe/blob/master/src/caffe/layers/elu_layer.cpp

TensorFlow's `EluGradOp::Compile` does not use alpha.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/tf2xla/kernels/elu_op.cc#L49

After checking this gradient code, I wonder if `selu_layer::backward_activation` is OK.

Comment (srk97) [reply tiny_dnn/activations/elu_layer.h]: Why would they change the backward pass and keep forward pass the same? :confused:

Comment (srk97) [reply tiny_dnn/activations/elu_layer.h]: I referred to the paper

**Comment by edgarriba** [comment]
@srk97 something went wrong with style format

File: test/test_serialization.h [hunk 3]
```diff
@@ -1006,7 +1008,7 @@ TEST(serialization, graph_model_and_weights) {
   fully_connected_layer f2(2, 2);
   softmax_layer a2(2);
   fully_connected_layer f3(2, 2);
-  elu_layer a3(2);
+  elu_layer a3((size_t)2);
```
Comment (edgarriba) [inline test/test_serialization.h]: for convention do:
`elu_layer a3(size_t(2))` or `elu_layer a3(static_cast<size_t>(2))`
